packages/frontend/src/app/modules/place/components/external-source-place-campaign-banner/external-source-place-campaign-banner.component.ts

```diff
@@ -0,0 +1,85 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import { Component, Input, OnDestroy, OnInit } from "@angular/core";
+
+import { CAMPAIGN_DEFAULT_NAME, PairingSources } from "@soliguide/common";
+
+import { CurrentLanguageService } from "../../../general/services/current-language.service";
+import { CAMPAIGN_LIST, Place, THEME_CONFIGURATION } from "../../../../models";
+import { PosthogService } from "../../../analytics/services/posthog.service";
+import { PosthogComponent } from "../../../analytics/components/posthog.component";
+import { Subscription } from "rxjs";
+
+// Define this constant outside the class
+const EXTERNAL_SOURCES: string[] = Object.values(PairingSources);
+
+@Component({
+  selector: "app-external-source-place-campaign-banner",
+  templateUrl: "./external-source-place-campaign-banner.component.html",
+  styleUrls: ["./external-source-place-campaign-banner.component.css"],
+})
+export class ExternalSourcePlaceCampaignBannerComponent
+  extends PosthogComponent
+  implements OnInit, OnDestroy
+{
+  public readonly THEME_CONFIGURATION = THEME_CONFIGURATION;
+  private readonly subscription = new Subscription();
+  @Input() public canEdit!: boolean;
+  @Input() public place!: Place;
+
+  public readonly CAMPAIGN_DATE_DEBUT_AFFICHAGE =
+    CAMPAIGN_LIST[CAMPAIGN_DEFAULT_NAME].dateDebutAffichage;
+
+  public shouldShowExternalBanner = false;
+  public routePrefix: string;
+
+  constructor(
+    private readonly currentLanguageService: CurrentLanguageService,
+    posthogService: PosthogService
+  ) {
+    super(posthogService, "update-campaign-banner");
+    this.routePrefix = this.currentLanguageService.routePrefix;
+  }
+
+  public ngOnInit(): void {
+    this.subscription.add(
+      this.currentLanguageService.subscribe(
+        () => (this.routePrefix = this.currentLanguageService.routePrefix)
+      )
+    );
+
+    const hasExternalSource =
+      Array.isArray(this.place.sources) &&
+      this.place.sources.some(
+        (s) => EXTERNAL_SOURCES.includes(s.name) && s.isOrigin === true
+      );
+
+    // Show banner for external sources if last update was before campaign and we're after campaign start
+    this.shouldShowExternalBanner =
+      hasExternalSource &&
+      this.place.updatedByUserAt < this.CAMPAIGN_DATE_DEBUT_AFFICHAGE &&
+      new Date() >= this.CAMPAIGN_DATE_DEBUT_AFFICHAGE;
```

Comment:
En fait, faut pas gérer la date
```suggestion
    this.shouldShowExternalBanner =
      Array.isArray(this.place.sources) &&
      this.place.sources.some(
        (s) => EXTERNAL_SOURCES.includes(s.name) && s.isOrigin === true
      );
```

Reply:
```suggestion
    this.shouldShowExternalBanner =
      this.place.sources?.some(
        (source) => EXTERNAL_SOURCES.includes(source.name) && source.isOrigin === true
      );
```

Pourquoi tu as besoin de checker si c'est un array ? Et mets plutôt des mots complets que des lettres dans tes fonctions

Reply:
C'était par sécurité ^^"
Yes pardon